Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2461,6 +2461,12 @@ void LocalDOMWindow::setLocation(LocalDOMWindow& activeWindow, const URL& comple
     if (completedURL.protocolIsJavaScript() && frameElement() && !frameElement()->document().contentSecurityPolicy()->allowJavaScriptURLs(aboutBlankURL().string(), { }, completedURL.string(), frameElement()))
         return;
 
+    // If the loader for activeWindow's frame (browsing context) has no outgoing referrer, set its outgoing referrer
+    // to the URL of its main frame's Document — which is the incumbent global object's associated Document.
+    // https://html.spec.whatwg.org/#location-object-navigate
+    if (activeWindow.frame()->loader().outgoingReferrer().isEmpty())
```

Comment:
`if (RefPtr activeFrame = activeWindow.frame(); activeFrame && activeFrame->loader().outgoingReferrer().isEmpty())` to be safe.